airflow/contrib/hooks/opsgenie_alert_hook.py

```diff
@@ -0,0 +1,173 @@
+    :param actions: Custom actions that will be available for the alert.
+    :type actions: list[str]
+    :param tags: Tags of the alert.
+    :type tags: list[str]
+    :param details: Map of key-value pairs to use as custom properties of the alert.
+    :type details: dict
+    :param entity: Entity field of the alert that is
+                    generally used to specify which domain alert is related to.
+    :type entity: str
+    :param source: Source field of the alert. Default value is
+                    IP address of the incoming request.
+    :type source: str
+    :param priority: Priority level of the alert. Default value is P3.
+    :type priority: str
+    :param user: Display name of the request owner.
+    :type user: str
+    :param note: Additional note that will be added while creating the alert.
+    :type note: str
+    :param proxy: Proxy to use to make the Opsgenie Alert API call
+    :type proxy: str
+    """
+    def __init__(self,
+                 http_conn_id=None,
+                 message="",
+                 alias=None,
+                 description=None,
+                 responders=None,
+                 visible_to=None,
+                 actions=None,
+                 tags=None,
+                 details=None,
+                 entity=None,
+                 source=None,
+                 priority=None,
+                 user=None,
+                 note=None,
+                 proxy=None,
+                 *args,
+                 **kwargs
+                 ):
+        super(OpsgenieAlertHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.api_key = self._get_api_key(http_conn_id)
+        self.message = message
+        self.alias = alias
+        self.description = description
+        self.responders = responders
+        self.visible_to = visible_to
+        self.actions = actions
+        self.tags = tags
+        self.details = details
+        self.entity = entity
+        self.source = source
+        self.priority = priority
+        self.user = user
+        self.note = note
+        self.proxy = proxy
+
+    def _get_api_key(self, http_conn_id):
+        """
+        Given a conn_id, return the api_key to use
+        :param http_conn_id: The conn_id provided
+        :type http_conn_id: str
+        :return: api_key (str) to use
+        """
+        if http_conn_id:
+            conn = self.get_connection(http_conn_id)
+            return conn.password
+        else:
+            raise AirflowException('Cannot get api_key: No valid conn_id '
+                                   'supplied')
+
+    def _build_opsgenie_payload(self):
+        """
+        Construct the Opsgenie JSON payload. All relevant parameters are combined here
+        to a valid Opsgenie JSON payload.
+
+        :return: Opsgenie payload (str) to send
+        """
+        payload = {}
+
+        payload['message'] = self.message
+        if self.alias:
```

Comment:
Could you create this in loop?
```python
for key in ["A".,"B"]:
	val = getattr(self, key)

	if val:
		payload[key] = val
```
Or maybe add a payload parameter to the constructor instead of copying the data? The user will provide data in the appropriate format. The operator will only perform appropriate validation. If it is light, it can be done in the constructor.

Reply:
Added payload parameter to hook constructor. Added your suggested loop for generating payload inside the operator from a new method during `execute`, so as to keep constructor light.